astropy/io/ascii/src/tokenizer.c

```diff
@@ -601,10 +601,15 @@ double str_to_double(tokenizer_t *self, char *str)
     {
         val = strtod(str, &tmp);
 
-        if (tmp == str || *tmp != 0)
+        if (tmp == str || *tmp != 0) {
+		    if (0 == strncmp(str, "nan", 3)) {
+				return NAN;
+		    }
```

Comment:
@mdmueller @taldcroft Any opinions on this change?  Should "inf" be supported too, and what about case-sensitivity.  Also it *looks* like `xstrtod` doesn't support inf/nan either, though I may be missing something (or maybe it's not supposed to?)


Also, looks like I need to fix a tab here, ironically.